python/lsst/meas/extensions/piff/piffPsfDeterminer.py

```diff
@@ -579,7 +580,7 @@ def _get_threshold(nth_order):
         if metadata is not None:
             metadata["spatialFitChi2"] = piffResult.chisq
             metadata["numAvailStars"] = len(stars)
-            metadata["numGoodStars"] = len(piffResult.stars)
+            metadata["numGoodStars"] = nUsedStars
```

Comment:
Is this `nUsedStars` - the number you computed before you called `fit` the second time? If no, you should calculate the `len` again. If yes, then you might as well assign the list you generated when you computed `nUsedStars` to a variable, say `usedStars` and reuse that from there onwards, say for `used_image_pos`.